xcube/core/dsio.py

```diff
@@ -538,37 +528,50 @@ def rimraf(path):
             pass
 
 
-def _get_path_or_store(path: str,
-                       client_kwargs: Dict[str, Any] = None,
-                       mode: str = None,
-                       root: str = None):
-    path_or_store = path
+def get_client_kwargs(kwargs) -> Tuple[dict, bool]:
     anon_mode = True
+    client_kwargs = dict(kwargs) if kwargs else {}
+    if 'client_kwargs' in client_kwargs:
+        client_kwargs = client_kwargs.pop('client_kwargs')
+    if 'endpoint_url' in client_kwargs:
+        client_kwargs['endpoint_url'] = client_kwargs.pop('endpoint_url')
+    if 'region_name' in client_kwargs:
+        client_kwargs['region_name'] = kwargs.pop('region_name')
+    if 'provider_access_key_id' in client_kwargs and 'provider_secret_access_key' in client_kwargs:
+        anon_mode = False
+        client_kwargs['aws_access_key_id'] = client_kwargs.pop('provider_access_key_id')
+        client_kwargs['aws_secret_access_key'] = client_kwargs.pop('provider_secret_access_key')
+    return client_kwargs, anon_mode
```

Comment:
`client_kwargs['aws_access_key_id']` and `client_kwargs['aws_secret_access_key']` no longer valid